src/com/amazon/ion/benchmark/IonSchemaUtilities.java

```diff
@@ -93,16 +101,25 @@ public static int parseConstraints(IonStruct value, String keyWord) throws IOExc
     }
 
     /**
-     * Return the value of annotations, this method is available when the annotations are required, if there are more constraints relative to the annotations, more logic needed to be added here.
+     * Return the information of annotations in a HashMap which includes the constraint and values of annotation. The constraint is the from set(optional | required |ordered)
      * @param value is the Ion struct which contain the current constraint field.
-     * @return an Ion List contains all annotations.
+     * @return a HashMap contains the information of annotations, both the constraint of the annotation and the value of the annotation.
      */
-    public static IonList getAnnotation(IonStruct value) {
-        IonList annotationList = null;
-        String annotationConstraint = value.get(keyWordAnnotations).getTypeAnnotations()[0];
-        if (annotationConstraint.equals(keyWordRequired)) {
-            annotationList = (IonList) value.get(keyWordAnnotations);
+    public static Map<String, Object> getAnnotation(IonStruct value) {
```

Comment:
Should this be `Map<String, IonValue>`?

Reply:
> Should this be `Map<String, IonValue>`?

It should be Map<String, Object>, because there one key:value pair which represents the constraint of annotation and the type of constraint is String which cannot be cast to IonValue, and the value of the other key:value pair which represents the real value of annotations is IonList.